lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -1597,6 +1606,12 @@ struct CanonicalizeConvertFromConvert
     if (auto view = dyn_cast<triton::ViewOp>(arg)) {
       if (isExpensiveView(view.getOperand().getType(), op.getType()))
         return failure();
+      // Part of values with dot operand encoding will be packed/unpacked as
+      // i32 elements when lowering to LLVM. To avoid errors, skip this folding
+      // when either the operand or result of view has a dot operand encoding.
```

Comment:
It's not clear in this comment why packing/unpacking as i32 is a problem.  I think the PR description is pretty clear.  Can we write something like that here?

Reply:
Good idea! Changed accordingly.